### PR TITLE
Refactor Focus to use browser-engine component.

### PR DESCRIPTION
**Pull request by pocmo**

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/utils/TabUtils.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    url: String,
+    source: SessionState.Source = SessionState.Source.NONE
+): Session {
+    // Creating a non-private Session as a workaround here. This needs to be an actual private
+    // session before we can land this. With a private session we are currently not able to save
+    // any tracking protection exceptions, see:
+    // https://bugzilla.mozilla.org/show_bug.cgi?id=1644156
+    // As a workaround we are currently removing all data manually, using Engine.clearData(), once
+    // a browsing session ends.
+    return Session(url, source = source, private = false)
```
Comment (pocmo) [inline app/src/main/java/org/mozilla/focus/utils/TabUtils.kt:20]: This was a workaround to get this working. But I think on `master` we could switch to `private = true` and then live with "broken" tracking protection exception storage until we get the GV fix for that.

**Review by jonalmeida — APPROVED** [review]
🎉 🎉 🎉 🎉 

Let's land it! :shipit:

File: app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt [hunk 2]
```diff
@@ -244,110 +189,209 @@ class BrowserFragment : WebFragment(), LifecycleObserver, View.OnClickListener,
...
+            SessionUseCases(components.store, components.sessionManager),
+            session.id,
+            ::viewportFitChanged,
+            ::fullScreenChanged
+        ), this, view)
+
+        contextMenuFeature.set(ContextMenuFeature(
+            parentFragmentManager,
+            components.store,
+            ContextMenuCandidate.defaultCandidates(
```
[the remark this answers is not in the document]
Comment (pocmo) [reply app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt:236]: Filed: https://github.com/mozilla-mobile/focus-android/issues/4707

File: app/src/main/java/org/mozilla/focus/ext/Session.kt [hunk 3]
```diff
@@ -52,13 +51,3 @@ var Session.savedGeckoSession: GeckoSession?
 var Session.savedWebViewState: Bundle?
```
[the remark this answers is not in the document]
Comment (pocmo) [reply app/src/main/java/org/mozilla/focus/ext/Session.kt:51]: Good point. Will do in the follow-up! :)

[the remark this answers is not in the document]
Comment (pocmo) [reply app/src/main/java/org/mozilla/focus/ext/Session.kt:51]: Well, actually, no, I will do it now. Need to update to make automation pass anyways. :)

**Comment by pocmo** [comment]
Need to look at some tests again to make this pass.

**Comment by pocmo** [comment]
🚥 🟢

**Pull request merged**